<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="ip" default="0.0.0.0"/>
  <arg name="port" default="4003"/>

  <arg name="acceleration_scale_factor" default="0.00000000186265"/>
  <arg name="gyroscope_scale_factor" default="0.000000000116415"/>
  <arg name="imu_hz" default="100"/>

  <group>
    <push-ros-namespace namespace="novatel_oem7"/>

    <node pkg="novatel_oem7_driver" exec="novatel_oem7_udp_sub" name="novatel_oem7_udp_sub" output="log">
      <param name="ip" value="$(var ip)"/>
      <param name="port" value="$(var port)"/>
    </node>

    <node pkg="novatel_oem7_driver" exec="novatel_sentence_to_navsatfix" name="novatel_sentence_to_navsatfix" output="log">
      <param name="acceleration_scale_factor" value="$(var acceleration_scale_factor)"/>
      <param name="gyroscope_scale_factor" value="$(var gyroscope_scale_factor)"/>
      <param name="imu_hz" value="$(var imu_hz)"/>
      <!--remap from="surface_speed" to="/sensing/vehicle_velocity_converter/twist_with_covariance"/-->
      <!--remap from="imu" to="/sensing/imu/imu_data"/-->
    </node>
  </group>
</launch>
